tools/aggsender_find_imported_bridge/aggsender_find_imported_bridge.go

```diff
@@ -0,0 +1,121 @@
+package main
+
+import (
+	"encoding/json"
+	"fmt"
+	"math/big"
+	"os"
+
+	"github.com/0xPolygon/cdk/agglayer"
+	"github.com/0xPolygon/cdk/aggsender/rpcclient"
+	"github.com/0xPolygon/cdk/aggsender/types"
+	"github.com/0xPolygon/cdk/bridgesync"
+	"github.com/0xPolygon/cdk/log"
+)
+
+const (
+	errLevelUnexpected         = 1
+	errLevelWrongParams        = 2
+	errLevelComms              = 3
+	errLevelNotFound           = 4
+	errLevelFoundButNotSettled = 5
+
+	base10         = 10
+	minimumNumArgs = 3
+)
+
+func unmarshalGlobalIndex(globalIndex string) (*agglayer.GlobalIndex, error) {
+	var globalIndexParsed agglayer.GlobalIndex
+	// First try if it's already decomposed
+	err := json.Unmarshal([]byte(globalIndex), &globalIndexParsed)
+	if err != nil {
+		bigInt := new(big.Int)
+		_, ok := bigInt.SetString(globalIndex, base10)
+		if !ok {
+			return nil, fmt.Errorf("invalid global index: %v", globalIndex)
+		}
+		mainnetFlag, rollupIndex, leafIndex, err := bridgesync.DecodeGlobalIndex(bigInt)
+		if err != nil {
+			return nil, fmt.Errorf("invalid global index, fail to decode: %v", globalIndex)
+		}
+		globalIndexParsed.MainnetFlag = mainnetFlag
+		globalIndexParsed.RollupIndex = rollupIndex
+		globalIndexParsed.LeafIndex = leafIndex
+	}
+	return &globalIndexParsed, nil
+}
+
+// This function find out the certificate for a deposit
+// It use the aggsender RPC
+func certContainsGlobalIndex(cert *types.CertificateInfo, globalIndex *agglayer.GlobalIndex) (bool, error) {
+	if cert == nil {
```

Comment:
Consider to make the `globalIndex` param a non-pointer

```suggestion
func certContainsGlobalIndex(cert *types.CertificateInfo, globalIndex agglayer.GlobalIndex) (bool, error) {
```

Also wdyt would it make sense to move this function into `CertificateInfo` struct and make it a method, something like this

```go
func (c *CertificateInfo) ContainsGlobalIndex(globalIndex agglayer.GlobalIndex) (bool, error) {
    ...
}
```